scripts/gha/it_workflow.py

```diff
@@ -0,0 +1,264 @@
+
+def test_start(token, issue_number, actor, commit, run_id):
+  """In PR, when start testing, add comment and label \"tests: in-progress\""""
+  github.add_label(token, issue_number, _LABEL_PROGRESS)
+  for label in [_LABEL_FAILED, _LABEL_SUCCEED]:
+    github.delete_label(token, issue_number, label)
+
+  comment = (_COMMENT_TITLE_PROGESS +
+             _get_description(actor, commit, run_id) +
+             _COMMENT_SUFFIX)
+  _update_comment(token, issue_number, comment)
+
+
+def test_progress(token, issue_number, actor, commit, run_id):
+  """In PR, when some test failed, update failure info and 
+  add label \"tests: failed\""""
+  log_summary = _get_summary_talbe(token, run_id)
+  if log_summary == 0:
+    return
+  else:
+    github.add_label(token, issue_number, _LABEL_FAILED)
+    comment = (_COMMENT_TITLE_PROGESS_FAIL +
+               _get_description(actor, commit, run_id) +
+               log_summary +
+               _COMMENT_SUFFIX)
+    _update_comment(token, issue_number, comment)
+
+
+def test_end(token, issue_number, actor, commit, run_id, new_token):
+  """In PR, when some test end, update Test Result Report and 
+  update label: add \"tests: failed\" if test failed, add label
+  \"tests: succeeded\" if test succeed"""
+  log_summary = _get_summary_talbe(token, run_id)
+  if log_summary == 0:
+    github.add_label(token, issue_number, _LABEL_SUCCEED)
+    comment = (_COMMENT_TITLE_SUCCEED +
+               _get_description(actor, commit, run_id) +
+               _COMMENT_SUFFIX)
+    _update_comment(token, issue_number, comment)
+  else:
+    github.add_label(token, issue_number, _LABEL_FAILED)
+    comment = (_COMMENT_TITLE_FAIL +
+               _get_description(actor, commit, run_id) +
+               log_summary +
+               _COMMENT_SUFFIX)
+    _update_comment(token, issue_number, comment)
+
+  github.delete_label(new_token, issue_number, _LABEL_PROGRESS)
+
+
+def test_report(token, actor, commit, run_id):
+  """Update (create if not exist) a Daily Report in Issue. 
+  If test failed, add label \"tests: failed\" and open the Issue, 
+  If test succeed, add label \"tests: succeeded\" and close the Issue.
+  The Issue with title _REPORT_TITLE and label _REPORT_LABEL:
+  https://github.com/firebase/firebase-cpp-sdk/issues?q=is%3Aissue+is%3Aclosed+label%3Anightly-testing
+  """
+  issue_number = _get_issue_number(token, _REPORT_TITLE, _REPORT_LABEL)
+  log_summary = _get_summary_talbe(token, run_id)
+  if log_summary == 0:
+#     github.delete_label(token, issue_number, _LABEL_FAILED)
+#     github.add_label(token, issue_number, _LABEL_SUCCEED)
+    github.close_issue(token, issue_number)
+    comment = (_COMMENT_TITLE_SUCCEED +
+               _get_description(actor, commit, run_id) +
+               _COMMENT_SUFFIX)
+    github.update_issue_comment(token, issue_number, comment)
+  else:
+#     github.delete_label(token, issue_number, _LABEL_SUCCEED)
```

Comment:
Because we are reusing the same issue, the label logic makes the issue grows longer. Plus, I feel opening/closing issue suggesting the success/failure states.
Anthony and I talked about it during today's sync meeting. He agreed with what we have right now. But, we can definitely make some changes later with feedbacks.
I also updated the design doc [When should we create a new Issue (pros and cons)](https://docs.google.com/document/d/1cQ3wcUuI7Zbaj8Mijz6LaTv11GAgrKHRP4nDPSvH7a8/edit?resourcekey=0-y5H3njKenX_Unet-oUPtkw#bookmark=id.6zxuggy4z0zy). Feel free to throw some comments